### ISSUE TITLE
opam2 list does not return correct error codes on solver failure

### ISSUE THREAD

**Issue by avsm** [ocaml/opam#3124]
This is seen in depext propagating through an exit code 99 on solver failure:

```
$ opam depext -u ppx_core.v0.9.3
# Detecting depexts using vars: arch=x86_64, os=linux, os-distribution=debian, os-version=9, os-family=debian
Fatal error: No solution for ppx_core.v0.9.3:
The following dependencies couldn't be met:
  - ppx_core -> stdio >= v0.9.1 -> base >= v0.9.4 -> sexplib >= v0.9.3 -> ocaml >= 4.05.0
    base of this switch, can't be changed (use `--unlock-base' to force)


Fatal error: exception Failure("Command failed: opam list --readonly --external  --resolve=ppx_core.v0.9.3 returned 99")
Removing intermediate container a3c1dee804e8
```

That `opam list` should probably fail with error 20 as its a solver failure.

**Comment by AltGr** [comment]
Indeed, `opam list --resolve` should be able to fail with 20

**Comment by avsm** [comment]
This is blocking CI quite a bit, since we cant distinguish between "valid" solver failures for ocaml versions

**Comment by AltGr** [comment]
This is fixed on master (by 472027079b8667949840cf88166aa85e0dd58452), but not in the beta5
